File: apps/cli/src/commands/start_server/base_app/controllers/get_metadata.ts
import os from 'os';
import definition from '@/definition';
import { Response } from '#/server/base/get_metadata';
import { Context } from '../constants';

const hostname = os.hostname();

export default async (ctx: Context) =>
  ctx.success<Response>({
    hostname,
    version: definition.VERSION,
  });

File: apps/flutter/lib/server/base/get_metadata.dart
import '../request.dart';
import '../../states/server.dart';

class Metadata {
  final String hostname;
  final String version;

  Metadata({required this.hostname, required this.version});

  factory Metadata.fromJSON(Map<String, dynamic> json) =>
      Metadata(hostname: json['hostname'], version: json['version']);
}

Future<Metadata> getMetadata(String? origin) async {
  final responseData = await httpGet(
    origin: origin ?? serverState.currentServer!.origin,
    path: "/base/metadata",
  );
  return Metadata.fromJSON(responseData);
}
